src/libcore/cell.rs

```diff
@@ -387,6 +388,33 @@ impl<T> Cell<T> {
         drop(old);
     }
 
+    /// Swaps the values of two Cells.
+    /// Difference with `std::mem::swap` is that this function doesn't require `&mut` reference.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(move_cell)]
+    /// use std::cell::Cell;
+    ///
+    /// let c1 = Cell::new(5);
+    /// let c2 = Cell::new(10);
+    /// c1.swap(&c2);
+    /// assert_eq!(10, c1);
+    /// assert_eq!(5, c2);
+    /// ```
+    #[inline]
+    #[unstable(feature = "move_cell", issue = "39264")]
+    pub fn swap(&self, other: &Self) {
+        if ptr::eq(self, other) {
+            return;
+        }
+        let temp = unsafe { mem::uninitialized() };
+        let o = other.replace(temp);
+        let s = self.replace(o);
+        other.set(s);
```

Comment:
The value in `other` is currently garbage and this will drop it.

You could do `mem::forget(other.replace(s))`, or this function could just be `unsafe { ptr::swap(self.as_ptr(), other.as_ptr()); }` (after the pointer equality check of course)